### DISCUSSION TITLE
Delete rows with files

### DISCUSSION THREAD

**Discussion by indadev** [pocketbase/pocketbase#997] [Q&A]
Hi
When I am deleting rows of a collection with files, the files are deleted fine but the folder structure stays in the storage folder.
That is normal? Im doing something wrong?

Thanks!
![image](https://user-images.githubusercontent.com/6991744/201323541-2090ce74-204e-4f4b-9f7a-15ae53efc2ac.png)

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it. When you upload a new record file, we create the following directory structure:
```
storage/collectionId/recordId/filename
storage/collectionId/recordId/thumbs_filename/...
```
Deleting a record should delete all its related files and thumbs (including the directory with the record id). The only remaining directory should be the parent collection id directory. It will be deleted with the collection deletions.

The only time when a record file will not be deleted is when you delete a collection **field** from the schema (this will be handled in https://github.com/pocketbase/pocketbase/issues/305).

**Reply by ganigeorgiev** [reply]
If the directories/files are not deleted for you, you coul try starting the application with `--debug` flag and check for console errors during the delete operation (it could be also a directory/file permission issue similar to the one in https://github.com/pocketbase/pocketbase/issues/305#issuecomment-1214212498).

**Reply by indadev** [reply]
In debug mode I dont see nothing
`[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE (`id`='files') OR (`name`='files') LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE (`id`='m30r32bxc73jn0d') OR (`name`='m30r32bxc73jn0d') LIMIT 1
[0.00ms] SELECT `files`.* FROM `files` WHERE (`files`.`id`='htbkfa897o39be3') AND (COALESCE('36jagv1gaqktm73', '') != COALESCE('', '')) LIMIT 1
[0.00ms] SELECT {{_collections}}.* FROM `_collections`
[0.00ms] DELETE FROM `files` WHERE `id`='htbkfa897o39be3'`

Before the delete
![image](https://user-images.githubusercontent.com/6991744/201333346-5fdf2a78-4433-417d-bd03-32b4940e2365.png)

After the delete
![image](https://user-images.githubusercontent.com/6991744/201333416-c6814430-721f-4166-81b7-9f0d498439bb.png)

The pb_data storage with the folders
![image](https://user-images.githubusercontent.com/6991744/201333604-a982bb52-e082-478d-aee5-24aa87f3ed9c.png)

This is the last one deleted that the folders exists

![image](https://user-images.githubusercontent.com/6991744/201333779-c607af91-8dfd-4d8d-91bd-fb79af3ad4b3.png)

**Reply by ganigeorgiev** [reply]
I'm not sure how to reproduce it. I've tested with both PocketBase v0.7.10 and v0.8.0-rc. Maybe this is something specific to Windows? Could you make sure that the windows user that started the PocketBase process has rights to read and modify the `storage` directory and its content?

**Reply by ganigeorgiev** [reply]
We also have integration tests that confirms that record deletion deletes the related files (including cascade deleted records) - https://github.com/pocketbase/pocketbase/blob/rc/apis/record_crud_test.go#L755-L778

**Reply by indadev** [reply]
I starter it in administrator mode and I have the same problem.
Only is happening when I upload a jpg or png that creates a thumbs.
If I upload a pdf file for example dont happen.
